newIDE/app/src/GameDashboard/LeaderboardAdmin/index.js

```diff
@@ -0,0 +1,835 @@
+  const isOnline = useOnlineStatus();
+  const windowWidth = useResponsiveWindowWidth();
+  const [isEditingName, setIsEditingName] = React.useState<boolean>(false);
+  const [isRequestPending, setIsRequestPending] = React.useState<boolean>(
+    false
+  );
+  const [newName, setNewName] = React.useState<string>('');
+  const [newNameError, setNewNameError] = React.useState<?string>(null);
+  const newNameTextFieldRef = React.useRef<?TextField>(null);
+  const [apiError, setApiError] = React.useState<?ApiError>(null);
+
+  const {
+    leaderboards,
+    listLeaderboards,
+    currentLeaderboardId,
+    createLeaderboard,
+    selectLeaderboard,
+    updateLeaderboard,
+    resetLeaderboard,
+    deleteLeaderboard,
+    deleteLeaderboardEntry,
+    displayOnlyBestEntry,
+    setDisplayOnlyBestEntry,
+    fetchLeaderboardEntries,
+    browsing: { entries },
+  } = React.useContext(LeaderboardContext);
+
+  const disableActions = React.useCallback(
+    (yesOrNo: boolean) => {
+      setIsRequestPending(yesOrNo);
+      onLoading(yesOrNo);
+    },
+    [onLoading]
+  );
+
+  const _updateLeaderboard = async (
+    i18n: I18nType,
+    payload: {|
+      name?: string,
+      sort?: LeaderboardSortOption,
+      playerUnicityDisplayChoice?: LeaderboardPlayerUnicityDisplayOption,
+    |}
+  ) => {
+    setNewNameError(null);
+    if (payload.name !== undefined && payload.name.length === 0) {
+      setNewNameError(
+        i18n._(
+          t`Please enter a name that is at least one character long and 50 at most.`
+        )
+      );
+      return;
+    }
+    disableActions(true);
+    setApiError(null);
+    try {
+      await updateLeaderboard(payload);
+      if (payload.name) setIsEditingName(false);
+    } catch (err) {
+      console.error(err);
+      setApiError({
+        action: payload.name
+          ? 'leaderboardNameUpdate'
+          : payload.sort
+          ? 'leaderboardSortUpdate'
+          : 'leaderboardPlayerUnicityDisplayChoiceUpdate',
+        message: payload.name ? (
+          <Trans>
+            An error ocurred when updating the name of the leaderboard, please
+            close the dialog, come back and try again.
+          </Trans>
+        ) : payload.sort ? (
+          <Trans>
+            An error ocurred when updating the sort direction of the
+            leaderboard, please close the dialog, come back and try again.
+          </Trans>
+        ) : (
+          <Trans>
+            An error ocurred when updating the display choice of the
+            leaderboard, please close the dialog, come back and try again.
+          </Trans>
+        ),
+      });
+    } finally {
+      disableActions(false);
+    }
+  };
+
+  const _listLeaderboards = React.useCallback(
+    () => {
+      const fetchAndHandleError = async () => {
+        disableActions(true);
+        setApiError(null);
+        try {
+          await listLeaderboards();
+        } catch (err) {
+          console.error(err);
```

Comment:
Prefer to always give some context, useful to understand errors when reading the console: 
```suggestion
          console.error(`Error while listing leaderboards:`, err);
```

Reply:
Will do!